### PR TITLE
tests: backport Sage's fixes to qa/suites/upgrade/jewel-x

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/19651

### CODE REVIEW

**Comment by smithfarm** [comment]
@liewegas `upgrade/jewel-x` is not in very good shape in kraken integration testing, so I'm going to try it again with these backports from master. Hope if I'm heading in the right direction here.

`upgrade/hammer-jewel-x` is in even worse shape btw.

**Comment by liewegas** [comment]
shouldn't need the mgr patches since mgr does nothing on kraken

**Comment by liewegas** [comment]
otherwise looks reasonable.. will need to run it and see!

**Comment by smithfarm** [comment]
@liewegas mgr cherry-pick dropped. Thanks for looking.

**Comment by smithfarm** [comment]
Jenkins re-test this please

**Comment by smithfarm** [comment]
@tchaikov @liewegas This does wonders for the `upgrade/jewel-x` suite - see http://tracker.ceph.com/issues/19009#note-72

The "failed to complete snap trimming before timeout" failures are unrelated to this PR (i'm pretty sure they're caused by #14597)

Please review.

**Comment by smithfarm** [comment]
@tchaikov Pinging for review.

File: qa/suites/upgrade/jewel-x/parallel/3-upgrade-sequence/upgrade-mon-osd-mds.yaml [hunk 1]
```diff
@@ -36,9 +36,3 @@ upgrade-sequence:
        daemons: [osd.2, osd.3]
        wait-for-healthy: false
        wait-for-osds-up: true
-   - exec:
```
Comment (tchaikov) [inline qa/suites/upgrade/jewel-x/parallel/3-upgrade-sequence/upgrade-mon-osd-mds.yaml]: @smithfarm i am not sure why we want to backport 468285b65d6bef2d86c80ebdfecf0920294ca5cd in two separated commits: 

- edd504c593bceac2059f475621177951c70a47e4
- a17afe0e1e50520996ff46bb0e81955fca11b823

**Comment by smithfarm** [comment]
@tchaikov Good catch! Fixed.

File: qa/suites/upgrade/jewel-x/parallel/3-upgrade-sequence/upgrade-mon-osd-mds.yaml (hunk 1 above)
Comment (smithfarm) [inline qa/suites/upgrade/jewel-x/parallel/3-upgrade-sequence/upgrade-mon-osd-mds.yaml]: I couldn't figure it out, either, so I just dropped both commits and redid the cherry-pick.

**Comment by smithfarm** [comment]
Passed an `upgrade/jewel-x` suite at http://tracker.ceph.com/issues/19009#note-76 and a rados suite at http://tracker.ceph.com/issues/19009#note-75

**Pull request merged**